### ISSUE TITLE
Memory issues caused by += with pytorch graph objects

### ISSUE THREAD

**Issue by jminuse** [jostmey/NakedAttention#1]
I tried running your code in a Kaggle notebook, but quickly ran out of RAM. I think the problem is that you are effectively building a computational graph that gets bigger every time you run `e_train += e_batch`, even though you only need the numerical value of `e_train` and you don't plan to backprop it. I think you can fix this by running `.detach()` on each pytorch object before you treat it as a plain numerical value. That way, the computational graph only contains a single batch, not the whole epoch. In my tests, this brings the memory usage back into the Kaggle bounds.

Here is the change I made:
```
-    e_train += e_batch/len(loader_train) # Accumulate average loss for this epoch
-    a_train += a_batch/len(loader_train) # Accumulate average accuracy for this epoch
+    e_train += e_batch.detach()/len(loader_train) # Accumulate average loss for this epoch
+    a_train += a_batch.detach()/len(loader_train) # Accumulate average accuracy for this epoch
```

**Comment by jostmey** [comment]
Fixed. Thanks!!
